File: apps/python-sync/app/models/sync_task.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text
from sqlalchemy.dialects.mysql import JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class SyncTask(Base):
    __tablename__ = "sync_tasks"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    task_no: Mapped[str] = mapped_column(String(64), unique=True, nullable=False)
    task_type: Mapped[str] = mapped_column(String(64), nullable=False)
    biz_type: Mapped[str] = mapped_column(String(64), nullable=False)
    biz_id: Mapped[str] = mapped_column(String(128), nullable=False)
    shop_id: Mapped[int] = mapped_column(Integer, nullable=False)
    platform_code: Mapped[str] = mapped_column(String(64), nullable=False)
    site_code: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(String(32), nullable=False)
    idempotency_key: Mapped[str] = mapped_column(String(128), unique=True, nullable=False)
    payload_json: Mapped[dict] = mapped_column(JSON, nullable=False)
    result_summary_json: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    retry_count: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    max_retry_count: Mapped[int] = mapped_column(Integer, nullable=False, default=3)
    last_error_code: Mapped[str | None] = mapped_column(String(64), nullable=True)
    last_error_message: Mapped[str | None] = mapped_column(Text, nullable=True)
    next_retry_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    accepted_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    finished_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_by: Mapped[str | None] = mapped_column(String(64), nullable=True)
    updated_by: Mapped[str | None] = mapped_column(String(64), nullable=True)
    created_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    updated_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)

File: apps/php-api/app/Models/SyncTask.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class SyncTask extends Model
{
    protected $table = 'sync_tasks';

    protected $fillable = [
        'task_no',
        'task_type',
        'biz_type',
        'biz_id',
        'shop_id',
        'platform_code',
        'site_code',
        'status',
        'idempotency_key',
        'payload_json',
        'result_summary_json',
        'retry_count',
        'max_retry_count',
        'last_error_code',
        'last_error_message',
        'next_retry_at',
        'accepted_at',
        'finished_at',
        'created_by',
        'updated_by',
    ];

    protected function casts(): array
    {
        return [
            'payload_json' => 'array',
            'result_summary_json' => 'array',
            'next_retry_at' => 'datetime',
            'accepted_at' => 'datetime',
            'finished_at' => 'datetime',
        ];
    }
}
